src/WebJobs.Script.WebHost/ExtensionsOptionContractResolver.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost
+{
+    public class ExtensionsOptionContractResolver : DefaultContractResolver
+    {
+        private static readonly HashSet<string> _skipSerialization = new HashSet<string>
```

Comment:
I'm wondering if it would be a viable solution for us to just use the IOptionsFormatter.Format method for each options instance we construct to get the reduced set of safe values? The whole point of the formatter is to produce a curated result w/o secrets, including only the "important" values. That's exactly what we need here.

Now the contract for Format is that it returns a string, but by convention this is always a json object (can verify for all existing extensions). In that case we can call Format to get the JSON, then turn around and reparse to JObject for inclusion in the sync payload.

Reply:
Hi @mathewc 
I agree with you. It is clean and safe solution. 

For the current implementation, I checked several options class on each extension.  I did some research for the current implementation of each extension.
 
* EventHubsOptions -> OK
* CosmosDBOptions -> OK
* KafkaOptions -> OK
* QueuesOptions, BlobsOptions, ServiceBusOptions -> not public (reflection?) ServiceBusOptions looks not follwing the format of host.json.
* Durable -> Not implement `IOptionsFormatter`

I'll double check the format of ServiceBusOptions by refelection. 

One of the reason why I didn't use this strategy, is the comment on the option. It says for logging. However, if all extensions follow the convention, I think it is clean / safer option. 

```
        /// <summary>
        /// Creates a string to be logged when the options are created from the IOptionsFactory.
        /// The returned value should be a JSON string with all secrets removed.
        /// </summary>
        /// <returns>A JSON string representing the options, with all secrets removed.</returns>
        string Format();
```

Reply:
We talked offline we'll go this direction. FYI: @bachuv @cgillum  https://github.com/Azure/azure-functions-durable-extension/issues/2061